src/EThree.ts

```diff
@@ -1,58 +1,100 @@
 import PrivateKeyLoader from './PrivateKeyLoader';
 import VirgilToolbox from './VirgilToolbox';
-import { CachingJwtProvider } from 'virgil-sdk';
+import { CachingJwtProvider, KeyEntryAlreadyExistsError } from 'virgil-sdk';
 import { VirgilPublicKey, Data } from 'virgil-crypto';
 import {
-    BootstrapRequiredError,
-    PasswordRequiredError,
+    RegisterRequiredError,
     EmptyArrayError,
     LookupError,
+    IdentityAlreadyExistsError,
+    PrivateKeyAlreadyExistsError,
+    MultipleCardsError,
 } from './errors';
 import { isWithoutErrors, isArray, isString } from './utils/typeguards';
 
 type EncryptVirgilPublicKeyArg = VirgilPublicKey[] | VirgilPublicKey;
 
+interface IEThreeOptions {
+    provider: CachingJwtProvider;
+    toolbox?: VirgilToolbox;
+    keyLoader?: PrivateKeyLoader;
+}
+
+const throwMultithreadError = (method: string) => {
```

Comment:
Even though this function is local to the module, the name is misleading. It is not a threading issue that this method is preventing. 
```suggestion
const throwIllegalInvocationError = (method: string) => {
```